ngraph/test/type_prop/roll.cpp

```diff
@@ -0,0 +1,139 @@
+
+    auto r = make_shared<op::v7::Roll>(arg, shift, axes);
+
+    EXPECT_EQ(r->get_output_element_type(0), element::f32);
+    EXPECT_TRUE(r->get_output_partial_shape(0).same_scheme(PartialShape{3, 3, 3}));
+}
+
+TEST(type_prop, roll_incorrect_axis_test)
+{
+    auto arg = make_shared<op::Parameter>(element::f32, Shape{3, 3});
+    auto shift = make_shared<op::Parameter>(element::i32, Shape{2});
+    auto axes = op::Constant::create(element::i64, Shape{2}, {0, 2});
+
+    try
+    {
+        auto r = make_shared<op::v7::Roll>(arg, shift, axes);
+        // Should have thrown, so fail if it didn't
+        FAIL() << "Unexpected pass with invalid axes and shift.";
+    }
+    catch (const NodeValidationFailure& error)
+    {
+        EXPECT_HAS_SUBSTRING(error.what(), std::string("Axes must be less than data tensor rank."));
+    }
+    catch (...)
+    {
+        FAIL() << "Check failed for unexpected reason";
+    }
+}
+
+TEST(type_prop, roll_incorrect_negative_axis_test)
+{
+    auto arg = make_shared<op::Parameter>(element::f32, Shape{3, 3});
+    auto shift = make_shared<op::Parameter>(element::i32, Shape{2});
+    auto axes = op::Constant::create(element::i64, Shape{2}, {0, -5});
+
+    try
+    {
+        auto r = make_shared<op::v7::Roll>(arg, shift, axes);
+        // Should have thrown, so fail if it didn't
+        FAIL() << "Unexpected pass with invalid axes and shift.";
+    }
+    catch (const NodeValidationFailure& error)
+    {
+        EXPECT_HAS_SUBSTRING(error.what(), std::string("Axes must be positive or equal to zero."));
+    }
+    catch (...)
+    {
+        FAIL() << "Check failed for unexpected reason";
+    }
+}
+
+TEST(type_prop, roll_axis_scalar_test)
+{
+    auto arg = make_shared<op::Parameter>(element::i32, Shape{3, 3, 4});
+    auto shift = op::Constant::create(element::i64, Shape{}, {5});
+    auto axes = make_shared<op::Parameter>(element::i32, Shape{3});
+
+    auto r = make_shared<op::v7::Roll>(arg, shift, axes);
+
+    EXPECT_EQ(r->get_output_element_type(0), element::i32);
+    EXPECT_TRUE(r->get_output_partial_shape(0).same_scheme(PartialShape{3, 3, 4}));
+}
+
+TEST(type_prop, roll_invalid_axes_check)
+{
+    auto arg = make_shared<op::Parameter>(element::f32, Shape{3, 3, 4, 1, 5});
+    auto shift = make_shared<op::Parameter>(element::i32, Shape{3});
+    auto axes = make_shared<op::Parameter>(element::i64, Shape{1});
+
+    try
+    {
+        auto r = make_shared<op::v7::Roll>(arg, shift, axes);
+        // Should have thrown, so fail if it didn't
+        FAIL() << "Unexpected pass with invalid axes and shift.";
+    }
+    catch (const NodeValidationFailure& error)
+    {
+        EXPECT_HAS_SUBSTRING(
+            error.what(),
+            std::string("If shift is a 1D vector, axes must be a 1D tensor of the same size."));
+    }
+    catch (...)
+    {
+        FAIL() << "Check failed for unexpected reason";
+    }
+}
+
+TEST(type_prop, roll_dynamic_shape)
```

Comment:
Add test cases with a mix of dynamic shape of `shift` and `axes`. For example, `shift` it dynamic while `axes` is not, and vice-versa.